contrib/win32/win32compat/misc.c

```diff
@@ -1725,3 +1725,177 @@ get_user_sid(char* name)
 
 	return ret;
 }
+
+/* builds session commandline. returns NULL with errno set on failure, caller should free returned string */
+char* build_session_commandline(const char *shell, const char* shell_arg, const char *command, int pty)
+{
+	enum sh_type { SH_CMD, SH_PS, SH_WSL_BASH, SH_CYGWIN, SH_OTHER } shell_type = SH_OTHER;
+	enum cmd_type { CMD_OTHER, CMD_SFTP, CMD_SCP } command_type = CMD_OTHER;
+	char *progdir = w32_programdir(), *cmd_sp = NULL, *cmdline = NULL, *ret = NULL, *p;
+	int len, progdir_len = (int)strlen(progdir);
+
+#define CMDLINE_APPEND(P, S)		\
+do {					\
+	int _S_len = (int)strlen(S);		\
+	memcpy((P), (S), _S_len);	\
+	(P) += _S_len;			\
+} while(0)
+
+	/* get shell type */
+	if (strstr(shell, "system32\\cmd"))
```

Comment:
It will help us in long run if we move this to a function that returns the enum.

Reply:
will retain this for now. IT should be easy to move it out if needed.